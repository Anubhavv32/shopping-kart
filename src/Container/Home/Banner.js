import React, { useEffect } from 'react'
import Slider from "react-slick";
import { connect } from 'react-redux';

import { fetchList } from '../../redux/action';

function Banner({bannersList, getBannersList}) {
  useEffect(() => {
    getBannersList();
  }, []);
  const settings = {
    dots: true,
    fade: true,
    infinite: true,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1,
    autoplay: true,
    autoplaySpeed: 1500
  };
  return (
    <div>
      <Slider {...settings}>
        {Array.isArray(bannersList) && bannersList.length ? bannersList.map( banner => {
          return (
          <div key={banner.id}>
            <img src={require(`../../${banner.bannerImageUrl}`).default} alt={banner.bannerImageAlt} />
          </div>
        )}) : null}
        </Slider>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    bannersList: state.bannersList
  };
};
const mapDispatchToProps = (dispatch) => {
  return {
    getBannersList: () => dispatch(fetchList('bannersJSON', 'GET_BANNERS_LIST')),
  };
};

export default React.memo(connect(mapStateToProps, mapDispatchToProps)(Banner));