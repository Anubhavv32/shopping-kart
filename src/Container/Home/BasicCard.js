import { Link } from "react-router-dom";
import React from "react";

export default function BasicCard({ category, index }) {
  console.log(category);
  return (
    <div
      className="card mb-3"
      key={category.id}
      //  style={{backgroundImage: `url(${img})`,
      // backgroundRepeat: 'no-repeat', backgroundSize: '100% 100%'}}
    >
      <div className="card-body">
        <div className="row w-100" style={index % 2 ? {flexDirection: "row-reverse"} : {flexDirection: "unset"}} >
          <div className="col-6">
            <img
              src={require(`../../${category.imageUrl}`)}
              alt=""
            />
          </div>
          <div className="col-6 text-center">
            <h5>{category.name}</h5>
            <p>{category.description}</p>
            <Link to="/products">
              <button className="btn btn-kart">Shop Online</button>
                </Link>
          </div>
        </div>
      </div>
    </div>
  );
}
