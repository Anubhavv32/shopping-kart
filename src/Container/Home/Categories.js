import React, { useEffect } from "react";
import BasicCard from "./BasicCard";
import { connect } from "react-redux";

import { getCategoriesList } from "../../redux/action";

function Categories({ getCategoriesList, categoriesList }) {
  useEffect(() => {
    getCategoriesList();
  }, []);
  return (
    <div className="category-banner">
      {Array.isArray(categoriesList) && categoriesList.length
        ? categoriesList.map((category, index) => {
            return <BasicCard category={category} key={category.id} index={index}/>;
          })
        : null}
    </div>
  );
}
const mapStateToProps = (state) => {
  return {
    categoriesList: state.categoriesList,
  };
};
const mapDispatchToProps = (dispatch) => {
  return {
    getCategoriesList: () => dispatch(getCategoriesList()),
  };
};
export default React.memo(
  connect(mapStateToProps, mapDispatchToProps)(Categories)
);
