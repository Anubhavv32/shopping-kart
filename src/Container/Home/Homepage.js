import React from 'react';
import Banner from './Banner';
import './homepage.css';
import Categories from './Categories';

function Homepage() {
  return (
    <div className='container'>
      <Banner />
      <Categories />
    </div>
  )
}

export default Homepage;