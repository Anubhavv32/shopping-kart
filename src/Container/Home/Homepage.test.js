import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import Homepage from './Homepage';
import Banner from './Banner';
import Categories from './Categories';
import BasicCard from './BasicCard';
const middlewares = []
const mockStore = configureStore(middlewares)

describe('MyComponent', () => {
  const store = mockStore({})
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Homepage debug store={store}/>);
    const componentBanner = shallow(<Banner debug store={store}/>);
    const componentCategories = shallow(<Categories debug store={store}/>);
    const componentBasicCard = shallow(<BasicCard debug store={store}/>);
    
    expect(component).toMatchSnapshot();
    
    expect(componentBanner).toMatchSnapshot();
    expect(componentCategories).toMatchSnapshot();
    expect(componentBasicCard).toMatchSnapshot();
  });
});