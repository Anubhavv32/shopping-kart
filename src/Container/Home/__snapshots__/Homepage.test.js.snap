// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div
  className="container"
>
  <Memo(Connect(Banner)) />
  <Memo(Connect(Categories)) />
</div>
`;

exports[`MyComponent should render correctly in "debug" mode 2`] = `
<ContextProvider
  value={null}
>
  <Banner
    debug={true}
    getBannersList={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`MyComponent should render correctly in "debug" mode 3`] = `
<ContextProvider
  value={null}
>
  <Categories
    debug={true}
    getCategoriesList={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`MyComponent should render correctly in "debug" mode 4`] = `""`;
